static/src/javascripts/projects/common/modules/commercial/creatives/facebook.js

```diff
@@ -1,63 +1,84 @@
     var scriptId = 'facebook-jssdk';
     var scriptSrc = '//connect.facebook.net/en_US/sdk/xfbml.ad.js#xfbml=1&version=v2.5';
     var adUnits = {
         mpu: { placementId: '180444840287_10154600557405288', adId: 'fb_ad_root_mpu' }
     };
     var facebookTpl;
+    var labelTpl;
 
     function Facebook($adSlot, params) {
         this.$adSlot = $adSlot;
         this.params  = params;
         facebookTpl || (facebookTpl = template(facebookStr));
+        labelTpl || (labelTpl = template(labelStr));
     }
 
     Facebook.prototype.create = function () {
-        window.fbAsyncInit || (window.fbAsyncInit = function() {
-            FB.Event.subscribe(
-                'ad.loaded',
-                function(placementID) {
-                    var interim = document.querySelector('[data-placementid="' + placementID + '"]');
-                    var ad = document.getElementById(interim.getAttribute('data-nativeadid'));
-                    if (ad) {
-                        fastdom.write(function () {
-                            ad.style.display = 'block';
-                        });
+        return new Promise(function (resolve) {
+            window.fbAsyncInit || (window.fbAsyncInit = function() {
+                FB.Event.subscribe(
+                    'ad.loaded',
+                    function(placementID) {
+                        var interim = document.querySelector('[data-placementid="' + placementID + '"]');
+                        var ad = document.getElementById(interim.getAttribute('data-nativeadid'));
+                        if (ad) {
+                            fastdom.write(function () {
+                                ad.style.display = 'block';
+                                resolve();
+                            });
+                        }
                     }
-                }
-            );
-
-            FB.Event.subscribe(
-                'ad.error',
-                function(errorCode, errorMessage, placementID) {
-                    reportError(new Error('Facebook returned an empty ad response'), {
-                        feature: 'commercial',
-                        placementID: placementID,
-                        errorMessage: errorMessage
-                    }, false);
-                }
-            );
-        });
+                );
 
-        var markup = facebookTpl(adUnits[this.params.placement]);
-        fastdom.write(function () {
-            this.$adSlot[0].insertAdjacentHTML('beforeend', markup);
+                FB.Event.subscribe(
+                    'ad.error',
+                    function(errorCode, errorMessage, placementID) {
+                        reportError(new Error('Facebook returned an empty ad response'), {
+                            feature: 'commercial',
+                            placementID: placementID,
+                            errorMessage: errorMessage
+                        }, false);
+                        resolve();
```

Comment:
Shouldn't this `reject`, so we can short circuit the `addLabel()` and `callback[...]` flows?